### ISSUE TITLE
Underline errors and warnings freezes spyder

### ISSUE THREAD

**Issue by impact27** [spyder-ide/spyder#11099]
### What steps will reproduce the problem?

1. Open `spyder/plugins/editor/widgets/codeeditor.py`
2. Open a parenthesis towards the beginning of the file without closing it
3. try to add and remove lines
3. `clear_extra_selections` takes ~5s for each line added or removed.called by: 
    - on_msg_received -> process_document_diagnostics -> handle_response -> process_diagnostics -> process_code_analysis -> cleanup_code_analysis
    - on_msg_received -> process_document_diagnostics -> handle_response -> process_diagnostics -> process_code_analysis -> highlight_selection -> set_extra_selections

## Versions

* Spyder version: 5.0.0.dev0 3055d36e8
* Python version: 3.7.5
* Qt version: 5.12.6
* PyQt5 version: 5.12.3
* Operating System: Darwin 19.2.0

### Dependencies

```
applaunchservices >=0.1.7    :  0.2.1 (OK)
atomicwrites >=1.2.0         :  1.3.0 (OK)
chardet >=2.0.0              :  3.0.4 (OK)
cloudpickle >=0.5.0          :  1.2.2 (OK)
diff_match_patch >=20181111  :  20181111 (OK)
intervaltree                 :  None (OK)
IPython >=4.0                :  7.11.0.dev (OK)
jedi =0.14.1                 :  0.14.1 (OK)
keyring                      :  None (OK)
nbconvert >=4.0              :  5.6.1 (OK)
numpydoc >=0.6.0             :  0.9.1 (OK)
pexpect >=4.4.0              :  4.7.0 (OK)
pickleshare >=0.4            :  0.7.5 (OK)
psutil >=0.3                 :  5.6.5 (OK)
pygments >=2.0               :  2.4.2 (OK)
pylint >=0.25                :  2.4.4 (OK)
pyls >=0.31.2;<0.32.0        :  0.31.2 (OK)
zmq >=17                     :  18.1.1 (OK)
qdarkstyle >=2.7             :  2.7 (OK)
qtawesome >=0.5.7            :  0.6.0 (OK)
qtconsole >=4.6.0            :  4.7.0.dev0 (OK)
qtpy >=1.5.0                 :  1.9.0 (OK)
sphinx >=0.6.6               :  2.2.1 (OK)
spyder_kernels >=1.8.1;<2.0.0:  1.9.0.dev0 (OK)
watchdog                     :  None (OK)
cython >=0.21                :  0.29.14 (OK)
matplotlib >=2.0.0           :  3.1.2 (OK)
numpy >=1.7                  :  1.17.4 (OK)
pandas >=0.13.1              :  0.25.3 (OK)
scipy >=0.17.0               :  1.3.3 (OK)
sympy >=0.7.3                :  1.4 (OK)
```

**Comment by ccordoba12** [comment]
What do you think we should do about this one @impact27?

**Comment by impact27** [comment]
I am not sure but this is bad enough that I would consider disabling "Underline errors and warnings"

**Comment by ccordoba12** [comment]
Fortunately, I think they are disabled by default. But you're right, this is really, really bad.

**Comment by ccordoba12** [comment]
I fixed this in 4.2.0
